src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Scaffold.kt

```diff
@@ -1446,8 +1448,10 @@ object Scaffold : Module("Scaffold", Category.WORLD, Keyboard.KEY_I, hideModule
     private fun switchBlockNextTickIfPossible(stack: ItemStack) {
         val player = mc.thePlayer ?: return
 
-        if (autoBlock !in arrayOf("Off", "Switch") && stack.stackSize <= 0) {
-            InventoryUtils.findBlockInHotbar()?.let {
+        if (autoBlock !in arrayOf("Off", "Switch") && ((earlySwitch && stack.stackSize <= amountBeforeSwitch) || stack.stackSize <= 0)) {
+            (if (earlySwitch) {InventoryUtils.findBlockStackInHotbarGreaterThan(amountBeforeSwitch)}
+             else{InventoryUtils.findBlockInHotbar()}
+            )?.let {
```

Comment:
This function has a terrible code style and this makes it even worse :/

1. the function has too many idents. And most of them are unnecessary. Please use guard statements.
2. too complex condition/control flow in condition: `autoBlock !in arrayOf("Off", "Switch") && ((earlySwitch && stack.stackSize <= amountBeforeSwitch) || stack.stackSize <= 0)` (the minimal amount before a switch occurs just changes, so you could represent this )
3. And this looks like an obfuscation technique:
```Kotlin
(if (earlySwitch) {InventoryUtils.findBlockStackInHotbarGreaterThan(amountBeforeSwitch)}
             else{InventoryUtils.findBlockInHotbar()}
            )?.let {
                TickScheduler += {
                    if (autoBlock == "Pick") {
                        player.inventory.currentItem = it - 36
                        mc.playerController.updateController()
                    } else {
                        serverSlot = it - 36
                    }
                }
```
If somebody reads that they have will use their entire mental capacity to understand what `it` is before they can even think about what the `.let { ... }` does

This is longer, but easier to follow:
````Kotlin
private fun switchBlockNextTickIfPossible(stack: ItemStack) {
    val player = mc.thePlayer ?: return
    
    if (autoBlock in arrayOf("Off", "Switch")) {
        return;
    }
    
    val switchThreshold = if (earlySwitch) amoundBeforeSwitch else 0
    
    if (stack.stackSize > switchThreshold) {
        return;
    }
    
    val newSlot = if (earlySwitch) {
        InventoryUtils.findBlockStackInHotbarGreaterThan(amountBeforeSwitch)
    } else {
        InventoryUtils.findBlockInHotbar()
    } ?: return

    TickScheduler += {
        if (autoBlock == "Pick") {
            player.inventory.currentItem = newSlot - 36
            mc.playerController.updateController()
        } else {
            serverSlot = newSlot - 36
        }
    }
}

Reply:
Surely he'll fix it, he told me that he doesn't know kotlin to well, and his main in java, so that might kinda explain his coding style..

Reply:
The thing is I did not code those hard to read things. I just appended only four lines in the spaghetti which worked the unreadable code there was.  LOL.

But I will surely fix it. Thanks for suggestion.